cli/azd/pkg/sqldb/sqldb.go

```diff
@@ -0,0 +1,60 @@
+package sqldb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/sql/armsql/v2"
+)
+
+// SqlDbService is the interface for the SqlDbService
+type SqlDbService interface {
+	ConnectionString(ctx context.Context, subId, rgName, serverName string, dbName string) (string, error)
+}
+
+type sqlDbClient struct {
+	credential azcore.TokenCredential
+	options    *arm.ClientOptions
+}
+
+// NewSqlDbService creates a new instance of the SqlDbService
+func NewSqlDbService(
+	credential azcore.TokenCredential,
+	options *arm.ClientOptions,
+) (SqlDbService, error) {
+	return &sqlDbClient{
+		credential: credential,
+		options:    options,
+	}, nil
+}
+
+// ConnectionString returns the connection string for the CosmosDB account
+func (c *sqlDbClient) ConnectionString(ctx context.Context, subId, rgName, serverName, dbName string) (string, error) {
+	clientFactory, err := armsql.NewClientFactory(subId, c.credential, c.options)
+	if err != nil {
+		return "", err
+	}
+
+	// get server fully qualified domain name
+	res, err := clientFactory.NewServersClient().Get(ctx, rgName, serverName, &armsql.ServersClientGetOptions{Expand: nil})
+	if err != nil {
+		return "", fmt.Errorf("failed getting server '%s' for resource group '%s'", serverName, rgName)
+	}
+
+	serverDomain := *res.Server.Properties.FullyQualifiedDomainName
```

Comment:
I think it's better to be more defensive here in derefencing the property.

`res.Server.Properties` may be `nil`
`res.Server.Properties.FullyQualifiedDomainName` may be nil.

I'd check for each one and provide a property error message. Otherwise, the nil panic will be hard to troubleshoot (we don't know which is unexpected, and why it's unexpected).